markdown_xblock/html.py

```diff
@@ -267,30 +277,118 @@ def get_editable_fields(self):
 
         return fields
 
+    @classmethod
+    def load_definition(cls, xml_object, system, location, id_generator):
+        """Load markdown content from file."""
+
+        filename = xml_object.get('filename')
+        pointer_path = "{category}/{url_path}".format(
+            category='markdown',
+            url_path=location.block_id.replace(':', '/')
+        )
+        base = path(pointer_path).dirname()
+        filepath = u"{base}/{name}.md".format(base=base, name=filename)
+
+        try:
+            with system.resources_fs.open(filepath, encoding='utf-8') as infile:
+                markdown = infile.read()
+
+                definition = {'data': markdown}
+
+                return definition
+
+        except ResourceNotFound as err:
+            msg = 'Unable to load file contents at path {0}: {1} '.format(
+                filepath, err)
+            raise Exception(msg)
```

Comment:
I still don't quite get this bit. You want to open the file, and if it doesn't exist, you want to throw the error up the stack. So what purpose does the `try/except` block serve here?

In an environment without `resources_fs`, you'd simply do:

```python
with open(filepath, encoding='utf-8') as infile:
  markdown = infile.read()
  definition = {'data': markdown}
  return definition
```
... which, for a `filepath` of `/tmp/foo.md` that does not exist, would raise

```
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/foo.md'
```

Is the error message for `ResourceNotFound` really so useless that you need to rewrap the whole thing with a new exception, copying the message?

Reply:
No need for the try/except block really, removed it now.